idurar/idurar-erp-crm
# Add API for Generating Invoice and Quotation Summary

# Context
This pull request introduces a new API endpoint that generates performance reports for both invoices and quotations relates with issues #226 . The reports provide valuable insights into the distribution of statuses over time, helping stakeholders make data-driven decisions.

# API Structure:
- Endpoint: `/api/{invoice / quote}/:type`
- Method: GET
- Parameters: type (optional): Defines the time period for the report. Possible values are "week," "month," or "year." If not specified, the default is "month."

## Response
```
{
    "success": true,
    "result": {
        "total": 11,
        "type": "week",
        "performance": [
            {
                "_id": "draft",
                "count": 6,
                "status": "draft",
                "percentage": 100
            },
            {
                "_id": "pending",
                "count": 1,
                "status": "pending",
                "percentage": 100
            },
            {
                "_id": "sent",
                "count": 4,
                "status": "sent",
                "percentage": 100
            }
        ]
    },
    "message": "Successfully found all invoices for the last week"
}
```

## How to Test:
To test the new API, follow these steps:
1. Clone this repository locally and switch to the feature branch.
2. Start the application and ensure it's running on your local server.
3. Use a tool like Postman or cURL to send a GET request to the API endpoint: Example: GET http://localhost:3000/api/{invoice / quote}/month
4. Replace "month" with "week" or "year" to generate reports for different time periods. Verify that the response contains the performance report in the specified time period, including the total count, type, and array of performance data for each status with count and percentage values.
